lib/TH/generic/THTensorMath.c

```diff
@@ -99,7 +99,7 @@ void THTensor_(nonzero)(THLongTensor *subscript, THTensor *tensor)
   long dim;
   long div = 1;
 #ifdef TH_REAL_IS_HALF
-#define IS_NONZERO(val) (TH_half2float(val)!=0)
+#define IS_NONZERO(val) (val.x!=TH_HALF_ZERO)
```

Comment:
I don't think this is equivalent -- previous code fails != check for 0.0f and -0.0f.  Current code only fails for 0.0f.  See e.g. https://github.com/numpy/numpy/blob/d9c1a1f91a3983469de7086ff589a6aeaa353b3e/numpy/core/src/npymath/halffloat.c#L50-L53